app/controllers/application_controller.rb

```diff
@@ -19,6 +19,19 @@ class ApplicationController < ActionController::Base
     end
   end
 
+  # check if this is a superAdmin user
+  def is_super_admin?  
+    if current_user && !current_user.policies.empty? && Policy.can?(MANAGE_USER, User, current_user, @current_user_policies)
+      true
+    else
+      false
+    end
+  end
+  
+  def authenticate_admin_user!
+    redirect_to new_user_session_path unless is_super_admin?
```

Comment:
If a non-admin user tries to access the admin dashboard it gets a redirect loop instead of an "unauthorized" message.